### PR TITLE
EE: Minor changes to syscall function

### PR DESCRIPTION
**Pull request by ssakash**
**Summary of changes**:

* Move interlace/progressive & Frame/Field detection code to ``SetGsCrt()`` instead of relying on SMODE2 registers.
* Add some nice enums to improve code readability on ``SYSCALL()``.

### CODE REVIEW

File: pcsx2/Counters.cpp [hunk 1]
```diff
@@ -34,7 +34,6 @@
 using namespace Threading;
 
 extern u8 psxhblankgate;
-extern bool gsIsInterlaced;
```
Comment (refractionpcsx2) [inline pcsx2/Counters.cpp]: I'm assuming that now won't compile. (what did I tell you about testing? :P )

Also below where you removed it from GS.h you will need to move that somewhere relevant as well.

Comment (ssakash) [inline pcsx2/Counters.cpp]: > I'm assuming that now won't compile. (what did I tell you about testing? :P )

Appveyor/TravisCI seems to be happy. :)

> Also below where you removed it from GS.h you will need to move that somewhere relevant as well.

I didn't remove anything from ``GS.h``. I assume you mean the ``gsIsInterlaced`` variable  ? It has been moved to get initialized at the SetGsCrt function.

Comment (refractionpcsx2) [inline pcsx2/Counters.cpp]: Sorry yes, you moved it, my bad :P

Comment (ssakash) [inline pcsx2/Counters.cpp]: > I'm guessing it's okay because GS.h is included in the counters?

Yeah, you're right.

**Comment by gregory38** [comment]
May I suggest to reset the video mode on reset (if not already done).
For the enum, it would be nice to add some numbers (=10, =20 ....) (or bigger step).

**Comment by willkuer** [comment]
Is there actually a reason for 632 enum entries but only five are actually handled in the switch? Does that mean that 99% of operations are not emulated. Or does it mean that these are handled somewhere else in the code where still the numerical flags 0x7F and so on are used and where it might be reasonable to use the same enum?

File: pcsx2/R5900OpcodeImpl.cpp [hunk 2]
```diff
@@ -883,8 +883,10 @@ void SYSCALL()
 
 	BIOS_LOG("Bios call: %s (%x)", R5900::bios[call], call);
```
Comment (gregory38) [inline pcsx2/R5900OpcodeImpl.cpp]: Does it compile in debug mode ? What happen if you enable the bios log trace (in the F10 menu)

**Comment by gregory38** [comment]
The kernel of the PS2 supports 256 syscalls. However, there are handled currently by the kernel code not by "us directly". When an enter syscall function is executed, we take the opportunity to hook it up to do extra jobs. I'm not sure we need all the entry is the enum. Even an HLE implementation will just use an array of function pointer. So we could reduce the enum to the useful value only.

File: pcsx2/R5900OpcodeImpl.cpp (hunk 2 above)
Comment (ssakash) [inline pcsx2/R5900OpcodeImpl.cpp]: Yes, it compiles fine and logging does work.

**Comment by ssakash** [comment]
> Does that mean that 99% of operations are not emulated. 

As Gregory said already, code is handled by the kernel. We only use this ``SYSCALL()`` to gather useful information (or) do some additional stuffs like in ``Deci2Call()``.

File: pcsx2/R5900OpcodeImpl.cpp [hunk 3]
```diff
@@ -882,10 +883,17 @@ void SYSCALL()
...
-					const char* inter = (cpuRegs.GPR.n.a0.UL[0] & 1) ? "Interlaced" : "Progressive";
-					const char* field = (cpuRegs.GPR.n.a2.UL[0] & 1) ? "FRAME" : "FIELD";
+
+	switch (static_cast<Syscall>(call)) {
+		case Syscall::SetGsCrt:
+		{
+					//Function "SetGsCrt(Interlace, Mode, Field)"
+					//Useful for fetching information of interlace/video/field display parameters of the Graphics Synthesizer
+
+					bool IsInterlaced = gsIsInterlaced = cpuRegs.GPR.n.a0.UL[0] & 1;
```
Comment (gregory38) [inline pcsx2/R5900OpcodeImpl.cpp]: `IsInterlaced` is useless now? You could use gsIsInterface directly, couldn't you?

File: pcsx2/R5900OpcodeImpl.cpp [hunk 4]
```diff
@@ -975,6 +973,17 @@ void SYSCALL()
 			}
 			break;
 
+		case Syscall::Deci2Call:
```
Comment (gregory38) [inline pcsx2/R5900OpcodeImpl.cpp]: move of code was kind of useless with naming enum

**Comment by gregory38** [comment]
Ok with the 2 small comments code is reviewed by me. I let's you update code accordingly.

**Comment by gregory38** [comment]
By the way @ssakash goes easy on tag. Otherwise github will have more tag than code source file. Goal is to make group not to replace title ;)

**Comment by refractionpcsx2** [comment]
> By the way @ssakash goes easy on tag.

yeh "Core" would have done :P

File: pcsx2/R5900OpcodeImpl.cpp [hunk 5]
```diff
@@ -882,10 +883,17 @@ void SYSCALL()
...
 
-	switch (call) {
-		case 2: {
-					const char* inter = (cpuRegs.GPR.n.a0.UL[0] & 1) ? "Interlaced" : "Progressive";
-					const char* field = (cpuRegs.GPR.n.a2.UL[0] & 1) ? "FRAME" : "FIELD";
+
+	switch (static_cast<Syscall>(call)) {
+		case Syscall::SetGsCrt:
+		{
+					//Function "SetGsCrt(Interlace, Mode, Field)"
```
Comment (turtleli) [inline pcsx2/R5900OpcodeImpl.cpp]: Indentation is rather strange for this case (issue existed before PR).

File: pcsx2/R5900OpcodeImpl.cpp (hunk 4 above)
Comment (ssakash) [inline pcsx2/R5900OpcodeImpl.cpp]: Some IDE's show the enum value as tooltip when hovering over the enum, I prefer having it sorted.

Comment (gregory38) [inline pcsx2/R5900OpcodeImpl.cpp]: Oh, kind of an useless feature. Name are better than number. But fine.

**Pull request merged**